### PR TITLE
BugFix - Which is newline terminated causing execution to fail

### PR DESCRIPTION
**Pull request by warmfusion**
Bugfix - Found that which returns newline's on strings which doesn't actually work as expected.

Don't know how this snuck through my testing on the host - we ensure that the newline is removed with a simple strip.

`Which` should never return multiple responses.
#### General

- [x] Update Changelog following the conventions laid out on [Keep A Changelog](http://keepachangelog.com/)

- [x] Update README with any necessary configuration snippets

~- [ ] Binstubs are created if needed~

- [ ] RuboCop passes

- [ ] Existing tests pass

### CODE REVIEW

File: bin/check-varnish-status.rb [hunk 1]
```diff
@@ -77,7 +77,7 @@ class CheckVarnishStatus < Sensu::Plugin::Check::CLI
   # Main Function
   def run
     # Keep a full reference for the varnish binary so sudo uses a full path
-    varnishadm = `which varnishadm 2>/dev/null`.to_s
+    varnishadm = `which varnishadm 2>/dev/null`.strip
```
Comment (majormoses) [inline bin/check-varnish-status.rb:80]: `chmop` would be a better choice

File: bin/metrics-varnish.rb [hunk 2]
```diff
@@ -62,7 +62,7 @@ def graphite_path_sanitize(path)
 
   def run
     # Keep a full reference for the varnish binary so sudo uses a full path
-    varnishstat_command = `which varnishstat 2>/dev/null`.to_s
+    varnishstat_command = `which varnishstat 2>/dev/null`.strip
```
Comment (majormoses) [inline bin/metrics-varnish.rb:65]: `chmop` would be a better choice

**Review by majormoses — CHANGES_REQUESTED** [review]
Good catch, I'd prefer we use `chmop` over `strip` because strip starts at the beginging of the string where chmop starts at the end in a minor net performance. You can read of the difference here: https://www.quora.com/What-is-the-difference-between-chomp-and-strip-in-Ruby

**Comment by rponcetn** [comment]
Hi guys, Why is it not yet merged? , I have the same error. Can not make the change by chomp or strip?

**Comment by majormoses** [comment]
@rponcetn I reviewed this and I suggested an alternative which is more performant and I have not heard back. Given the lack of response I am gonna merge (to give credit) and change it myself.

**Review by majormoses — APPROVED** [review]

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-varnish/versions/1.1.1

**Pull request merged**